### DISCUSSION TITLE
Allow <noscript> in <style scoped>

### DISCUSSION THREAD

**Discussion by Pandapip1** [vuejs/rfcs#559] [General]
## Summary

Currently, there is no way to apply styles depending on if JS is supported within components, since <style> tags aren't allowed in components. I have a hamburger menu that I want to display on mobile devices. However, it requires JS to operate, and I'd rather have the navbar perpetually extended than have a nonfunctional hamburger menu. This requires styles wrapped in a <noscript> tag.

## Proposed API

```html
<style scoped>
    /* regular CSS */
    <noscript>
        /* CSS that gets wrapped in <noscript> tag *
    </noscript>
</style>
```

**Comment by darrylnoakes** [comment]
I assume you're using SSR in some form?
Because nothing will render if you're using pure client-side rendering and JS is disabled.

**Reply by Pandapip1** [reply]
I'm using vitepress which uses SSR, yes.
